README.md

```
@@ -1,61 +1,201 @@
+- ```apim_control_plane```: Control Plane profile for API publishing, management, and key management(if key manager is not separated).
+- ```apim_tm```: Traffic Manager profile for throttling and rate limiting.
+- ```apim_km```: Key Manager Profile for API manager.
+- ```apim_common```: Shared logic, parameters, and files used by all profiles.
+
+![Module architecture](docs/images/module_architecture.png "Module architecture")
+
+The run stages for Puppet are described in `<puppet_environment>/manifests/site.pp`, and they are of the order Main -> Custom.
+
+Each Puppet module manifest contains the following .pp files.
+* Main
+    * ```params.pp```: Contains all the parameters necessary for the main configuration and template.
```

markdownlint-cli2 (0.18.1)

Unordered list indentation
Expected: 2; Actual: 4

(MD007, ul-indent)